src/plugins/intel_cpu/tests/functional/single_layer_tests/group_convolution.cpp

```diff
@@ -196,12 +200,51 @@ class GroupConvolutionLayerCPUTest : public testing::WithParamInterface<groupCon
                 ngraph::builder::makeGroupConvolution(paramOuts[0], netType, kernel, stride, padBegin,
                                                       padEnd, dilation, padType, convOutChannels, numGroups));
         function = makeNgraphFunction(netType, params, groupConv, "groupConvolution");
+        if (inputShape.second[0][1] / numGroups <= 16) {
+            isSmallInputChannel = true;
+        }
     }
 };
 
 TEST_P(GroupConvolutionLayerCPUTest, CompareWithRefs) {
     SKIP_IF_CURRENT_TEST_IS_DISABLED()
 
+   // Skip tests for brgconv convolution where kernel size = 1x1
+    if (priority[0] == "brgconv_avx512" || priority[0] == "brgconv_avx512_amx") {
+        bool is_1x1 = true;
+        for (const auto &i : kernel) {
+            if (i != 1) {
+                is_1x1 = false;
+                break;
+            }
+        }
+        if (is_1x1) {
+            GTEST_SKIP() << "Disabled test due to the brgconv does not support 1x1 convolution kernel." << std::endl;
+        }
+        if (isSmallInputChannel) {
+            GTEST_SKIP() << "Disabled test due to the brgconv does not support small input channel GroupConvolution on amx." << std::endl;
+        }
+    }
+    if (priority[0] == "brgconv_avx512" && isBF16) {
+        if (!(with_cpu_x86_bfloat16() && with_cpu_x86_avx512_core_vnni())) {
```

Comment:
Could you please provide the context of this change.
Is it right that oneDNN will not use brgconv when those parameters are set?
I mean if it is 1x1 and input channels are small, etc
If so, and the problem is that we are just forcing the brgconv primitive in scope of the tests,
I think it will more convenient and much more clear if we will just have separate test instances for brgconv / amx cases and then we can avoid those lots of "if" check which makes the test hardly readable to be honest.
What do you think, is it possible to just have separate instances?

Reply:
**About the context**
about 1x1 please check https://github.com/openvinotoolkit/oneDNN/blob/6df930dab5ab0a7dfaea6100acd03b479e2fa0a8/src/cpu/x64/jit_brgemm_conv_utils.cpp#L1833
about small input channel https://github.com/openvinotoolkit/oneDNN/blob/6df930dab5ab0a7dfaea6100acd03b479e2fa0a8/src/cpu/x64/jit_brgemm_conv_utils.cpp#L2069  and I found the condition is alread changed after migrate to onednn2.7, I will check the test case again
**Separate instances**
I will try to separate test case for "brgconv / amx cases"

Reply:
@EgorDuplensky  I already separate the test case, please help to review